HDDS-9520. Merge some TestOzoneManagerHA subclasses

adoroszlai opened the pull request:
## What changes were proposed in this pull request?

Multiple `TestOzoneManagerHA` subclasses test OM HA setup.  Each subclass needs to start/stop its own cluster, which takes about 30 seconds.  Some of the classes could be merged to reduce the number of clusters.

https://issues.apache.org/jira/browse/HDDS-9520

## How was this patch tested?

Before:

```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 41.235 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAWithACL
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 62.146 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAKeyDeletion
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 45.204 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAWithFailover
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 39.471 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAMetrics
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 154.38 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAWithData
[INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 155.955 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAMetadataOnly
```

After:

```
[INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 210.821 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAWithData
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 164.196 s - in org.apache.hadoop.ozone.om.TestOzoneManagerHAMetadataOnly
```

hemantk-12 commented on hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithData.java:
```diff
@@ -551,4 +571,577 @@ private String createMultipartUploadPartKey(OzoneBucket ozoneBucket,
+  }
+
+  @Test
+  void testRunAllTests() throws Exception {
```
nit: I think individual tests can run independently.

hemantk-12 reviewed (approved):
Thanks for working on this @adoroszlai.

It is a good change and will reduce the CI/CD time.

It is simple merging of multiple test classes to single class to eliminate the test setup time. No change is test as such. LGTM.

adoroszlai commented:
Thanks @hemantk-12 for the review.

I've updated the patch: instead of grouping tests by "metadata" and "data", now they are split based on whether they stop/restart OMs or not.  This way we can avoid `@BeforeEach`/`@AfterEach` steps for one of the test classes and split the ACL test method without performance impact.

hemantk-12 reviewed (approved):
LGTM.

adoroszlai commented:
Thanks @hemantk-12 for the reviews.
